### DISCUSSION TITLE
Stable combination of library versions

### DISCUSSION THREAD

**Discussion by Qilanya** [FilledStacks/stacked#743] [Q&A]
Hello all,

I was very excited when I found the stacked architecture for building my idea of a flutter app.
I tried to follow the very basic setup [tutorial](https://www.filledstacks.com/post/new-setup-for-flutter-stacked-state-management/)  from filledstacks.com with the simple startup and second view navigation.

Unfortunately I ran into issues with both the example code and the build_runner due to some dependencies not agreeing with each other or with the sdk on the system. 

So my question is which combination of versions of the following libraries/sdks work well together in your experience?
- Java JDK
- Android SDK
- Flutter 
- stacked
--stacked_services
- stacked_generator
- build_runner
- Android Studio/Visual Studio Code

I'd really appreciate your help on this, as I had trouble finding a setup that doesn't run into some error or another.
Thank you.

**Comment by FilledStacks** [comment]
Hey, I haven't had many issues with the latest setup. As long as everything is on the latest it should work. 

Stacked is only dependent on Flutter so you can ignore the following things in your comparison:
- Java SDK
- Android SDK
- Android Studio / Visual Studio Code

Use the [Stacked Tools](https://stacked.filledstacks.com/docs/Ecosystem/stacked-cli/) and generate an app and see if that works.

**Reply by Qilanya** [reply]
Hello Dane,

thank you very much for the quick reply.

I have installed the tools as per the linked documentation and I have two more questions/issues about that.

On running "stacked create app my_app" I get a few **errors that certain files are not found** and the corresponding **commands failed** (see output below). It is generating an app, but it is incomplete (e.g. no locator function - I can run the build_runner manually in Android studio and it generates it though, so the issue is only during creation).

Is there an additional step required to be able to run it e.g. in Android studio? 
The current version of my_app is missing something like an Android manifest according to Android studio. Was able to fix that by copying the android folter of another Flutter app created straight in Android Studio into the generated my_app folder.  So the test app is up and running and I'm excited to build my own on this framework.

Log:
Running flutter create my_app ...
Command failed. Das System kann die angegebene Datei nicht finden.

Add Stacked Magic ...
Created File: 'my_app\test/viewmodels/home_viewmodel_test.dart'
Created File: 'my_app\test/helpers/test_helpers.dart'
Created File: 'my_app\stacked.config.json'
Created File: 'my_app\README.md'
Created File: 'my_app\lib/main.dart'
Created File: 'my_app\lib/ui/setup/setup_dialog_ui.dart'
Created File: 'my_app\lib/ui/setup/setup_bottom_sheet_ui.dart'
Created File: 'my_app\lib/ui/common/ui_helpers.dart'
Created File: 'my_app\lib/ui/common/app_strings.dart'
Created File: 'my_app\lib/ui/common/app_colors.dart'
Created File: 'my_app\lib/ui/bottom_sheets/notice_sheet/notice_sheet.dart'
Created File: 'my_app\lib/ui/bottom_sheets/notice_sheet/notice_sheet_viewmodel.dart'
Created File: 'my_app\lib/ui/dialogs/info_alert/info_alert_dialog.dart'
Created File: 'my_app\lib/ui/dialogs/info_alert/info_alert_viewmodel.dart'
Created File: 'my_app\lib/ui/views/home/home_view.dart'
Created File: 'my_app\lib/ui/views/home/home_viewmodel.dart'
Created File: 'my_app\lib/ui/views/startup/startup_viewmodel.dart'
Created File: 'my_app\lib/ui/views/startup/startup_view.dart'
Created File: 'my_app\lib/app/app.dart'
Created File: 'my_app\lib/enums/dialog_type.dart'
Created File: 'my_app\lib/enums/bottom_sheet_type.dart'
Created File: 'my_app\pubspec.yaml'
Running flutter format . -l 80 in my_app/...
Command failed. Das System kann die angegebene Datei nicht finden.

Running flutter pub get in my_app/...
Command failed. Das System kann die angegebene Datei nicht finden.

Running flutter pub run build_runner build --delete-conflicting-outputs in my_app/...
Command failed. Das System kann die angegebene Datei nicht finden.

Running flutter format . -l 80 in my_app/...
Command failed. Das System kann die angegebene Datei nicht finden.

**Reply by FilledStacks** [reply]
There's no additional commands required. Can you run `flutter doctor -v` and paste the output below. If I understand the message correctly, Flutter is not installed properly on your system.

**Reply by Qilanya** [reply]
I had checked that and had some issues earlier. I installed Java somewhat late, after Flutter as I noticed it was missing. And Android Studio is pointing out that the plugins for Flutter and Dart can be installed at some address, though in the Settings of Android Studio they are marked as installed. I did re-install Android Studio once though, maybe that messed it up.

flutter doctor -v
[√] Flutter (Channel stable, 3.3.5, on Microsoft Windows [Version 10.0.19044.2130], locale de-CH)
    • Flutter version 3.3.5 on channel stable at C:\Users\xxx\Desktop\App_Projekt\flutter2022\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d9111f6402 (6 days ago), 2022-10-19 12:27:13 -0700
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\xxx\AppData\Local\Android\Sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.3.32929.385
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.72.2)
    • VS Code at C:\Users\xxx\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.50.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.2130]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 106.0.5249.119
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 106.0.1370.47

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**Reply by Qilanya** [reply]
I just realized I had an underscore in the flutter path when it should not contain any special characters.  Maybe that's a problem. Will check that again.

**Reply by Qilanya** [reply]
Still the same errors during creation (like mentioned in the log above) - and the android folder is missing from the generated apps...   I also reset Android Studio to reinstall the plugins, checked the Path variables again. I am out of ideas on what could be the issue here..

And now I can´t even fix the project from within Android Studio.
Running the build runner gives this log:

[INFO] Generating build script...
[INFO] Generating build script completed, took 604ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 106ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 788ms

[INFO] Running build...
[INFO] Running build completed, took 34ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 86ms

[SEVERE] mockito:mockBuilder on test/helpers/test_helpers.dart (cached):

Null check operator used on a null value
[SEVERE] Failed after 138ms
pub finished with exit code 1

**Reply by Qilanya** [reply]
I also tried "flutter create" and that seemed to work. The created app does not have an android sdk assigned automatically in the android studio project structure settings though and the flutter plugins folder in libraries is empty. Not sure whether that is fine like that. It does run fine on an emulator as is, so seems to have no issues.

**Reply by FilledStacks** [reply]
If the `flutter create` command doesn't generate your android folder then the stacked_tools won't either. We just run flutter_create after creating our template. 

Very strange. I rarely see issues with flutter installation. Did you do anything special for installation? or move files around after?

**Reply by Qilanya** [reply]
The Android folder does exist with flutter create - just in the project structure in Android studio the sdk is not filled. 
I did reinstall flutter in a path without special characters later,  but the folder was not generated by stacked even before that. And trying to resolve this I also reinstalled Android Studio and such. 
What baffles me is that the app currently generated by flutter create is complete,  but the one by stacked create app is incomplete because it can't locate flutter? Really strange.

**Reply by Qilanya** [reply]
I'll try to set up stacked on a different laptop in the next couple of days - it might really just be an issue with mine. Thank you for the pointers so far.

**Reply by FilledStacks** [reply]
Happy to give pointers. It would have been better if it could be solved. We do nothing special. We execute flutter directly in your command line, no extra special things.

**Reply by Qilanya** [reply]
So I gave it another go on a different machine with the same end result unfortunately. Back on my own I laptop I uninstalled everything, cleaned the Path variable and pub caches, etc. After restarting, these were the steps taken:
- verify requirements for Flutter as per https://docs.flutter.dev/get-started/install/windows
- i.e. re-install Git, check Powershell version
- get Flutter source code from Git this time and clone it into proposed folder C:\src
- add flutter\bin to user Path variables in system environment
- re-opened console and ran flutter doctor > this executed pub get/upgrade and the usual checks > still missing Android Sdk / Studio
- installed Android Studio
- installed the plugins in Android Studio (Flutter, Dart)
- re-ran flutter doctor > all green after accepting Android licenses (see screenshot)
- ran dart pub global activate stacked_tools > no errors, just a message asking to add the path to the system environment variables (done - see screenshot)
- re-started the laptop and tested the create commands

>> flutter create worked as expected
>> stacked create still shows the errors and the app folder is missing files (e.g. android folder, pubspeck.lock)

Attached a few more screenshots for you. Do you see anything out of the ordinary?
![doctor](https://user-images.githubusercontent.com/92530599/198978969-5c7efb8d-d309-4b38-b560-bde02e87d890.PNG)
![flutter_app](https://user-images.githubusercontent.com/92530599/198978974-14d80b30-f7e1-41a6-bb83-1f6cc31f648f.PNG)
![flutter_create](https://user-images.githubusercontent.com/92530599/198978977-c455c048-190e-44c9-a39d-b37c838506cd.PNG)
![stacked_app](https://user-images.githubusercontent.com/92530599/198978980-63412c9b-2c9a-444d-a9b7-7b7932bcd6e3.PNG)
![stacked_create](https://user-images.githubusercontent.com/92530599/198978981-6cd1595b-ea70-46ab-beb3-ab1bf972aca6.PNG)
![sysenv_path_variable](https://user-images.githubusercontent.com/92530599/198978984-5936ccd2-6f00-46ed-8631-4606b0896c65.PNG)

**Reply by FilledStacks** [reply]
Very strange. I have no idea why it's doing that. I'll see if we can maybe do a check and understand why that's happening. 

The process might not have the correct permissions? I'll have to be able to reproduce on my side to fix it.
